diff --git a/dist/cjs/UtilityParser.js b/dist/cjs/UtilityParser.js
index 74ba4df933620cae8b415567218b0c3739d17073..94c12785938c62ae8d4583807abcb578b4982668 100644
--- a/dist/cjs/UtilityParser.js
+++ b/dist/cjs/UtilityParser.js
@@ -43,6 +43,7 @@ const opacity_1 = require("./resolve/opacity");
 const shadow_1 = require("./resolve/shadow");
 const transform_1 = require("./resolve/transform");
 const pointer_events_1 = __importDefault(require("./resolve/pointer-events"));
+const user_select_1 = __importDefault(require("./resolve/user-select"));
 class UtilityParser {
     constructor(input, config = {}, cache, device, reactNativeVersion) {
         var _a;
@@ -343,6 +344,11 @@ class UtilityParser {
             if (style)
                 return style;
         }
+        if (this.consumePeeked(`select-`)) {
+            style = (0, user_select_1.default)(this.rest);
+            if (style)
+                return style;
+        }
         h.warn(`\`${this.rest}\` unknown or invalid utility`);
         return null;
     }
diff --git a/node_modules/twrnc/dist/cjs/resolve/user-select.js b/dist/cjs/resolve/user-select.js
new file mode 100644
index 0000000000000000000000000000000000000000..dfb932b83c1887d4417a6510db68e0aaf59c073c
--- /dev/null
+++ b/dist/cjs/resolve/user-select.js
@@ -0,0 +1,12 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const helpers_1 = require("../helpers");
+const ALLOWED_VALUES = [`auto`, `text`, `none`, `contain`, `all`];
+function userSelect(value) {
+    if (!ALLOWED_VALUES.includes(value))
+        return null;
+    return (0, helpers_1.complete)({
+        userSelect: value,
+    });
+}
+exports.default = userSelect;
diff --git a/dist/esm/UtilityParser.js b/dist/esm/UtilityParser.js
index 1e01913a9ac5bf76c45654d281f55c18bb25a69a..95f3f2a05fa911d2550bf3bf58c74f32d30c8f44 100644
--- a/dist/esm/UtilityParser.js
+++ b/dist/esm/UtilityParser.js
@@ -15,6 +15,7 @@ import { opacity } from './resolve/opacity';
 import { shadowOpacity, shadowOffset } from './resolve/shadow';
 import { origin, rotate, scale, skew, transformNone, translate, } from './resolve/transform';
 import pointerEvents from './resolve/pointer-events';
+import userSelect from './resolve/user-select';
 export default class UtilityParser {
     constructor(input, config = {}, cache, device, reactNativeVersion) {
         var _a;
@@ -315,6 +316,11 @@ export default class UtilityParser {
             if (style)
                 return style;
         }
+        if (this.consumePeeked(`select-`)) {
+            style = userSelect(this.rest);
+            if (style)
+                return style;
+        }
         h.warn(`\`${this.rest}\` unknown or invalid utility`);
         return null;
     }
diff --git a/node_modules/twrnc/dist/esm/resolve/user-select.d.ts b/dist/esm/resolve/user-select.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..4aab165afaa8cc1a51e9f0f041d0ef64ee67c5db
--- /dev/null
+++ b/dist/esm/resolve/user-select.d.ts
@@ -0,0 +1,2 @@
+import type { StyleIR } from '../types';
+export default function userSelect(value: string): StyleIR | null;
diff --git a/node_modules/twrnc/dist/esm/resolve/user-select.js b/dist/esm/resolve/user-select.js
new file mode 100644
index 0000000000000000000000000000000000000000..81f2d032f2a8411760e0a44092fd4bb96af4960b
--- /dev/null
+++ b/dist/esm/resolve/user-select.js
@@ -0,0 +1,9 @@
+import { complete } from '../helpers';
+const ALLOWED_VALUES = [`auto`, `text`, `none`, `contain`, `all`];
+export default function userSelect(value) {
+    if (!ALLOWED_VALUES.includes(value))
+        return null;
+    return complete({
+        userSelect: value,
+    });
+}
